### ISSUE TITLE
Stop statsmodels.tsa.x13.x13_arima_analysis from creating temporary files

### ISSUE THREAD

**Issue by kuritzen** [statsmodels/statsmodels#8277]
Im using package statsmodels in python. To be precise I am using the function statsmodels.tsa.x13.x13_arima_analysis(). When I use it, it creates a lot if temporary files in folder C:\Users\username\AppData\Local\Temp. I dont want it to create these files, how can i prevent this? Or, if it would be possible to choose where these out files appear that would be great.

I know that the function has a log parameter and i have tried setting this equal to false without any success. 

Thanks in advance,
Felix Kuritzén

**Comment by josef-pkt** [comment]
AFAIK, the files are required to communicate with the x13 program. I don't think there is another way.

the current `x13_arima_analysis` uses `tempfile` and has the following:

```
    # write it to a tempfile
    # TODO: make this more robust - give the user some control?
    ftempin = tempfile.NamedTemporaryFile(delete=False, suffix='.spc')
    ftempout = tempfile.NamedTemporaryFile(delete=False)
```

**Comment by kuritzen** [comment]
Yes i noticed that too. It would be nice though to be able to choose log_filepath. They can create an own "tempfile" module and let us users decide where to put the logs.

Maybe that is to come by reading line 
 `# TODO: make this more robust - give the user some control?`

**Comment by josef-pkt** [comment]
https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile
there is a `dir` option to choose the directory

So, this looks relatively easy to add a `dir` or `tempdir` option to the x13 function.

Do you want to try and make a PR?

**Comment by kuritzen** [comment]
Perfect, i will try it out locally and then edit online. Will do this next week though.
